tool/src/demo_gen/terminus.rs

```diff
@@ -0,0 +1,410 @@
+                            self.ctx.errors.push_error(format!("Found usage of disabled type {type_name}"))
+                        }
+
+                        type_name
+                    },
+                    _ => unreachable!("Unknown primitive type {:?}", param_type),
+                };
+
+                out_param(type_name);
+            }
+            Type::Opaque(o) => {
+                let op = o.resolve(self.ctx.tcx);
+                let type_name = self.ctx.formatter.fmt_type_name(o.tcx_id.into());
+
+                let all_attrs = &o.resolve(self.ctx.tcx).attrs;
+                if all_attrs.disable {
+                    self.ctx.errors.push_error(format!("Found usage of disabled type {type_name}"))
+                }
+                let attrs = &all_attrs.demo_attrs;
+
+                if attrs.external {
+                    out_param(type_name.into());
+                    return;
+                }
+
+                // We need to find a constructor that we can call.
+                // Piggybacking off of the #[diplomat::attr(constructor)] macro for now as well as test attributes in attrs.rs
+                let mut usable_constructor = false;
+
+                for method in op.methods.iter() {
+                    if usable_constructor {
+                        break;
+                    }
+
+                    let method_attrs = &method.attrs.demo_attrs;
+                    usable_constructor |= method_attrs.default_constructor;
+                    if let Some(diplomat_core::hir::SpecialMethod::Constructor) =
+                        method.attrs.special_method
+                    {
+                        usable_constructor |= true;
+                    }
+
+                    if usable_constructor {
+                        self.terminus_info
+                            .imports
+                            .insert(self.ctx.formatter.fmt_import_statement(
+                                &type_name.clone(),
+                                false,
+                                "../".into(),
+                            ));
+
+                        let mut child = MethodDependency::new(
+                            self.get_constructor_js(type_name.to_string(), method),
+                        );
+
+                        let call = self.evaluate_constructor(method, &mut child);
+                        node.params.push(ParamInfo {
+                            js: call,
+                            label: "".into(),
+                            type_name: String::default(),
+                        });
+                    }
+                }
+                if !usable_constructor {
+                    self.ctx.errors.push_error(format!("You must set a default constructor for the opaque type {}, as it is required for the function {}. Try adding #[diplomat::demo(default_constructor)] above a method that you wish to be the default constructor.", op.name.as_str(), node.method_js));
```

Comment:
"or disable it in the demo backend"

also nit: i'd linewrap this, in Rust you can linewrap strings with indentation and it works as long as you precede each newline with a backslash